### ISSUE TITLE
Ruby 3.1 support

### ISSUE THREAD

**Issue by oskarpearson** [DataDog/datadog-lambda-rb#70]
## Overview

The AWS lambda stack only supports Ruby 2.7 _when running in the "Amazon Linux" environment_. However, AWS lambda *also* supports running Docker Images.

These Docker images can be any version of Ruby - including version 3.1.

This gem seems to only support Ruby 2.7 - while Ruby 3.0 was released in December 2020, and Ruby 3.1 was released in December 2021.

For those of us choosing to use a recent docker-based image (instead of the S3 file based lambda image where only 2.7 is supported) we are unable to use this gem.

Attempting to use this gem with Ruby 3.1 gives the following error, due to the way that later versions of Ruby handle keyword arguments.

## Expected Behavior

This gem would work with Ruby 2.7 and later versions of Ruby such as 3.1

## Actual Behavior

The gem doesn't support tracing with Ruby 3.1.

## Steps to Reproduce the Problem

  1. Install Ruby 3.1
  1. Try and use the `Datadog::Lambda.wrap` method
  1. You will receive a message saying "wrong number of arguments (given 2, expected 1) (ArgumentError)"

## Specifications

  - Datadog Lambda Layer version: Not relevant - using a docker image
  - Ruby version: ruby:3.1.3-slim docker image
  
## Stacktrace
  
  ```
/Users/oskar/.rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/ddtrace-1.7.0/lib/datadog/tracing.rb:17:in `trace': wrong number of arguments (given 2, expected 1) (ArgumentError)
	from /Users/oskar/.rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/datadog-lambda-2.16.0/lib/datadog/lambda/trace/ddtrace.rb:36:in `wrap_datadog'
	from /Users/oskar/.rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/datadog-lambda-2.16.0/lib/datadog/lambda/trace/listener.rb:51:in `on_wrap'
	from /Users/oskar/.rbenv/versions/3.1.3/lib/ruby/gems/3.1.0/gems/datadog-lambda-2.16.0/lib/datadog/lambda.rb:57:in `wrap'
	from (irb):39:in `block in <top (required)>'
  ```

A monkey-patch such as the following fixes this issue, but there may be other similar issues. The `options` parameter needs to be prefixed with `**`

```
module Datadog
  module Trace
    class <<self
      def wrap_datadog(options = {}, &block)
        unless Datadog::Tracing.enabled?
          Datadog::Utils.logger.debug 'dd-trace unavailable'
          return block.call
        end

        Datadog::Tracing.trace('aws.lambda', **options) do |_span|
          block.call
        end
      end
    end
  end
end
```

**Comment by sfirrin** [comment]
Hi @oskarpearson and @djpate, thanks for opening this issue. We are going to investigate what would be required for us to support 3.1. In the meantime could you please open a feature request with Datadog support to help us prioritize?

**Comment by oskarpearson** [comment]
Thanks @sfirrin - I've opened datadog support request #1030550

**Comment by oskarpearson** [comment]
The response I got was:

>I've reached out to our internal team and unfortunately supporting Ruby v3.1 for lambda is not currently on their roadmap.

**Comment by oskarpearson** [comment]
@sfirrin and @djpate I've submitted a two-character PR that should fix Ruby 3.1 compatibility without any change to 3.1 compatibility... of course "without guarantee" from Datadog.

PS Datadog folks - Ruby 3.2 support is expected from AWS sometime early this year - so you may want to ensure that's on your roadmap. Check the PR for details.

**Comment by duncanista** [comment]
Closing as per #76
